src/libraries/System.Private.Xml/tests/XmlWriter/DisposeTests.cs

```diff
@@ -121,5 +123,161 @@ public static void XmlWriterDisposeWorksWithDerivingClasses()
+            using (var stream1 = new MemoryStream())
+            using (var stream2 = new MemoryStream())
+            using (var stream3 = new MemoryStream())
+            {
+                using (var asyncWriter = XmlWriter.Create(stream1, settings))
+                {
+                    await asyncWriter.WriteStartDocumentAsync();
+                    await asyncWriter.WriteStartElementAsync(string.Empty, "root", null);
+                    await asyncWriter.WriteStartElementAsync(null, "test", null);
+                    await asyncWriter.WriteAttributeStringAsync(string.Empty, "abc", string.Empty, "1");
+                    await asyncWriter.WriteEndElementAsync();
+                    await asyncWriter.WriteEndElementAsync();
+                }
+
+                await using (var asyncWriter = XmlWriter.Create(stream2, settings))
+                {
+                    await asyncWriter.WriteStartDocumentAsync();
+                    await asyncWriter.WriteStartElementAsync(string.Empty, "root", null);
+                    await asyncWriter.WriteStartElementAsync(null, "test", null);
+                    await asyncWriter.WriteAttributeStringAsync(string.Empty, "abc", string.Empty, "1");
+                    await asyncWriter.WriteEndElementAsync();
+                    await asyncWriter.WriteEndElementAsync();
+                }
+
+                settings.Async = false;
+                using (var syncWriter = XmlWriter.Create(stream3, settings))
+                {
+                    syncWriter.WriteStartDocument();
+                    syncWriter.WriteStartElement(string.Empty, "root", null);
+                    syncWriter.WriteStartElement(null, "test", null);
+                    syncWriter.WriteAttributeString(string.Empty, "abc", string.Empty, "1");
+                    syncWriter.WriteEndElement();
+                    syncWriter.WriteEndElement();
+                }
+
+                Assert.Equal(stream1.GetBuffer(), stream2.GetBuffer());
+                Assert.Equal(stream2.GetBuffer(), stream3.GetBuffer());
+
+                Assert.Equal(@"<?xml version=""1.0"" encoding=""utf-8""?><root><test abc=""1"" /></root>", ReadAsString(stream1.GetBuffer(), stream1.Length));
+                Assert.Equal(@"<?xml version=""1.0"" encoding=""utf-8""?><root><test abc=""1"" /></root>", ReadAsString(stream2.GetBuffer(), stream2.Length));
+                Assert.Equal(@"<?xml version=""1.0"" encoding=""utf-8""?><root><test abc=""1"" /></root>", ReadAsString(stream3.GetBuffer(), stream3.Length));
+            }
+        }
+
+        private static string ReadAsString(byte[] bytes, long length) => Encoding.UTF8.GetString(bytes, 0, (int)length);
+
+        [Fact]
+        public static async Task AsyncWriterDispose_ShouldCall_FlushAsyncWriteAsyncOnly_TextWriter()
+        {
+            using (var sw = new AsyncOnlyWriter())
+            await using (var writer = XmlWriter.Create(sw, new XmlWriterSettings() { Async = true }))
```

Comment:
ditto

Reply:
and anywhere where applies

Reply:
`The .ConfigureAwait(false) usage can/should be removed in the tests.` https://github.com/dotnet/runtime/pull/35233#discussion_r412256260